pinot-core/src/test/java/com/linkedin/pinot/queries/PregeneratedHllTest.java

```diff
@@ -1,153 +0,0 @@
-        writer.create(updatedSchema, newAvroFile);
-        while (avroReader.hasNext()) {
-          GenericRecord record = avroReader.next();
-          GenericData.Record newRecord = new GenericData.Record(updatedSchema);
-          for (Schema.Field field : fields) {
-            newRecord.put(field.name(), record.get(field.name()));
-          }
-          newRecord.put(hllColumnName, HllUtil.singleValueHllAsString(log2m, record.get(column)));
-          writer.append(newRecord);
-        }
-      }
-    }
-    return newAvroFile;
-  }
-
-  private void createLoadSegment(String segmentName, File destinationDir)
-      throws Exception {
-    final String column = "column1";
-    File newAvroFile = new File(destinationDir, segmentName + ".avro");
-    createAvroWithHll(newAvroFile, testHelper.AVRO_DATA, column, hllConfig.getHllLog2m());
-    helper = testHelper.buildHllSegment(testHelper.tableName, newAvroFile.getAbsolutePath(), testHelper.TIME_COLUMN,
-        testHelper.TIME_UNIT, segmentName, hllConfig, false);
-    testHelper.loadSegment(helper.getSegmentDirectory());
-  }
-
-  @Test
-  public void testHllOnPregeneratedColumn()
-      throws Exception {
-    File destinationDir = null;
-    try {
-      destinationDir = Files.createTempDirectory(PregeneratedHllTest.class.getName()).toFile();
-
-      final String segmentName = "pregeneratedHllSegment";
-      createLoadSegment(segmentName, destinationDir);
-
-      // Init Query Executor
-      final List<AvroQueryGenerator.TestSimpleAggreationQuery> aggCalls = new ArrayList<>();
-      aggCalls.add(
-          new AvroQueryGenerator.TestSimpleAggreationQuery("select fasthll(column1_hll) from " + testHelper.tableName +
-              " where column1 > 100000 limit 0", 0.0));
-      aggCalls.add(new AvroQueryGenerator.TestSimpleAggreationQuery(
-          "select distinctcounthll(column1) from " + testHelper.tableName +
-              " where column1 > 100000 limit 0", 0.0));
-
-      // we give high value of 1000 because this test also runs with log2m different from distinctCounthll
-      // function. There is no guarantee about counts or accuracy. I want to make sure these are valid values
-      ApproximateQueryTestUtil
-          .runApproximationQueries(testHelper.queryExecutor, helper.getSegmentName(), aggCalls, 1000,
-              testHelper.serverMetrics);
-
-      Double val1 = (Double)ApproximateQueryTestUtil
-          .runQuery(testHelper.queryExecutor, Arrays.asList(helper.getSegmentName()), aggCalls.get(0),
-              testHelper.serverMetrics);
-      // 5000, because we know
-      Assert.assertTrue(val1 > 5000);
-      // add one more segment. Goal to verify that combine works correctly
-      // across multiple segments. Not interested in exact values
-      final String secondSegmentName = "segment2";
-      createLoadSegment(secondSegmentName, destinationDir);
```

Comment:
This part of the code also verifies that combining hll results across segments works correctly. We need to preserve that test functionality (wherever we think is appropriate location)